### PR TITLE
Remove some more explicit SearchResponse use in tests Pt2

### PR DESCRIPTION
**Pull request by piergm**
The title says the whole story part 2

### CODE REVIEW

**Comment by piergm** [comment]
@elasticmachine update branch

File: x-pack/plugin/transform/src/internalClusterTest/java/org/elasticsearch/xpack/transform/integration/TransformProgressIT.java [hunk 1]
```diff
@@ -214,11 +214,12 @@ private TransformProgress getProgress(Function function, SearchRequest searchReq
         final AtomicReference<TransformProgress> progressHolder = new AtomicReference<>();
         final AtomicReference<Exception> exceptionHolder = new AtomicReference<>();
 
-        SearchResponse response = client().search(searchRequest).actionGet();
-        function.getInitialProgressFromResponse(
-            response,
-            new LatchedActionListener<>(ActionListener.wrap(progressHolder::set, exceptionHolder::set), latch)
-        );
+        assertResponse(client().search(searchRequest), response -> {
```
Comment (original-brownbear) [inline x-pack/plugin/transform/src/internalClusterTest/java/org/elasticsearch/xpack/transform/integration/TransformProgressIT.java:217]: NIT: no need for curly brackets, just a single statement

File: x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/actions/email/EmailAttachmentTests.java [hunk 2]
```diff
@@ -205,8 +205,15 @@ public void testThatEmailAttachmentsAreSent() throws Exception {
                     throw e;
                 }
             }
-            assertNotNull(searchResponse);
-            assertHitCount(searchResponse, 1);
+            try {
```
Comment (original-brownbear) [inline x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/actions/email/EmailAttachmentTests.java:208]: I guess you could just move the try down one line and save. the `if` in the `finally` :) If it's null we don't need to release it and don't even get to the second line.

**Review by original-brownbear — APPROVED** [review]
LGTM 2 small cleanup suggestions, thanks!

**Comment by piergm** [comment]
@elasticmachine update branch

File: x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/actions/email/EmailAttachmentTests.java (hunk 2 above)
Comment (piergm) [reply x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/actions/email/EmailAttachmentTests.java:208]: Done! I updated also `WatchMetadataTests` that had a similar try-finally block

**Pull request merged**